splitio/sync/segment.py

```diff
@@ -174,13 +175,153 @@ def synchronize_segments(self, segment_names = None, dont_wait = False):
         """
         if segment_names is None:
             segment_names = self._split_storage.get_segment_names()
-            
+
         for segment_name in segment_names:
             self._worker_pool.submit_work(segment_name)
         if (dont_wait):
             return True
         return not self._worker_pool.wait_for_completion()
-    
+
+    def segment_exist_in_storage(self, segment_name):
+        """
+        Check if a segment exists in the storage
+
+        :param segment_name: Name of the segment
+        :type segment_name: str
+
+        :return: True if segment exist. False otherwise.
+        :rtype: bool
+        """
+        return self._segment_storage.get(segment_name) != None
+
+class LocalSegmentSynchronizer(object):
+    """Localhost mode segment synchronizer."""
+
+    _DEFAULT_SEGMENT_TILL = -1
+
+    def __init__(self, segment_folder, split_storage, segment_storage):
+        """
+        Class constructor.
+
+        :param segment_folder: patch to the segment folder
+        :type segment_folder: str
+
+        :param split_storage: Split Storage.
+        :type split_storage: splitio.storage.InMemorySplitStorage
+
+        :param segment_storage: Segment storage reference.
+        :type segment_storage: splitio.storage.SegmentStorage
+
+        """
+        self._segment_folder = segment_folder
+        self._split_storage = split_storage
+        self._segment_storage = segment_storage
+        self._segment_sha = {}
+
+    def synchronize_segments(self, segment_names = None):
+        """
+        Loop through given segment names and synchronize each one.
+
+        :param segment_names: Optional, array of segment names to update.
+        :type segment_name: {str}
+
+        :return: True if no error occurs. False otherwise.
+        :rtype: bool
+        """
+        _LOGGER.info('Synchronizing segments now.')
+        if segment_names is None:
+            segment_names = self._split_storage.get_segment_names()
+
+        return_flag = True
+        for segment_name in segment_names:
+            if not self.synchronize_segment(segment_name):
+                return_flag = False
+
+        return return_flag
+
+    def synchronize_segment(self, segment_name, till=None):
+        """
+        Update a segment from queue
+
+        :param segment_name: Name of the segment to update.
+        :type segment_name: str
+
+        :param till: ChangeNumber received.
+        :type till: int
+
+        :return: True if no error occurs. False otherwise.
+        :rtype: bool
+        """
+        try:
+            fetched = self._read_segment_from_json_file(segment_name)
+            if fetched == {}:
```

Comment:
wouldn't be better to check for None?

Reply:
Possible, I am checking if the returned is {} then synchronize_segments() method will return False

Reply:
fixed